### ISSUE TITLE
UTM closes when starting any VM

### ISSUE THREAD

**Issue by Egleyser** [utmapp/UTM#2398]
When starting any VM downloaded directly from the UTM website, the application simply closes. My perception is that it closes and does not crash the app.

* I did the test with my iPad8 Gen, with the UTM in the same version the problem is the same.
* I tried the images of Debian Minimal and ArchLinux.

2.0.25
iOS 14.2
Tests performed on iPad8Gen iPhone6s
Both with Jailbreak
Via repository

**Comment by osy** [comment]
Why do you say it does not crash the app?

**Comment by Egleyser** [comment]
because it doesn't close abruptly, like a bug or something, it just minimizes it. If I try to create a VM it starts and I can even see the boot screen, for example. However, importing the VM image will not.

**Comment by osy** [comment]
I see, can you get a debug log?

**Comment by Egleyser** [comment]
Please, can you send me instructions on how to send these logs?

**Comment by osy** [comment]
So when you created the issue, there was a template text that I will copy paste here

> For all issues, _including_ crashes, you should attach a debug log. Open UTM, and open the settings for the VM you wish to launch. Near the bottom of the configuration options (or at the top of the `QEMU` page) is `Debug Log`. Turn it on and save the VM. After you experience the issue, quit the VM and re-launch UTM. Open the VM settings again and select `Export Log...` and attach it here.

**Comment by Egleyser** [comment]
Oops, I’m sorry, I didn’t pay attention, it’s attached.
[debug.log](https://github.com/utmapp/UTM/files/6191964/debug.log)

**Comment by osy** [comment]
`qemu-system: -accel tcg,tb-size=32: allocate 33554432 bytes for jit buffer: Invalid argument` that's the issue. The JIT buffer by default takes your memory size (128M) and divides that by 4. Seems like this is too small here. Either increase the size (512M) or use the advanced setting for JIT buffer size and bump that up to something like 128

**Comment by Egleyser** [comment]
Well, I would already comment here, I found it very strange because I am doing the tests on the 2 devices, both on the iPad and on the iphone6s, both gave the same thing, after the edits you had suggested. Regarding the tweaks, I'm not sure, because I use minimal adjustments on the device. I'm having tweak flame to give you the list of settings I have but this option is not working. Any suggestions for tweak that may cause this or any that I can list for me to send here?

**Comment by osy** [comment]
Just to confirm you used the `cydia.getutm.app` repository correct?

**Comment by Egleyser** [comment]
Yes, I used this repo.

**Comment by osy** [comment]
Hmm I can't immediately think of a reason. I know other people have run into this problem (jailbroken) but I don't have many jailbroken devices to test with and my own devices work without issue but I don't have any tweaks installed.

**Comment by Egleyser** [comment]
I will continue the tests, I will use the iPad that has less tweaks installed for those who know or remove all tweaks or disable them until I find the cause of the problem. Anything I keep you informed here, I thank you in advance.

**Comment by Egleyser** [comment]
I will leave my default jailbreak, delete all settings and even then UTM does not start any VM. Very strange. I will restart, sign with Altstore, and see how it will behave.

**Comment by Egleyser** [comment]
Got it: D
At first sideloaded with Altstore / Altdeamon and rebooted the device, after the restart it still did not go up the Debian VM, I checked the logs and checked that it could be something with memory, it decreased and the VM went up. I redid the Jailbreak with Unc0ver, I tested it again and the VM went up perfectly on both devices perfectly fouled just sideloaded by the altstore. Based on my tests I believe it is something with a certificate / signature, this calls for a more in-depth investigation.

**Comment by osy** [comment]
Do you mean the IPA installed by altstore works while jailbroken? Or did you reinstall from cydia?

**Comment by Egleyser** [comment]
Exactly. Installed directly from the cydia repositories, the VM does not start.

**Comment by tristan-k** [comment]
> because it doesn't close abruptly, like a bug or something, it just minimizes it. If I try to create a VM it starts and I can even see the boot screen, for example. However, importing the VM image will not.

Same for me. I'm on an ipad pro (2020) with a jailbroken iOS 14.3.

**Comment by Egleyser** [comment]
> > because it doesn't close abruptly, like a bug or something, it just minimizes it. If I try to create a VM it starts and I can even see the boot screen, for example. However, importing the VM image will not.
> 
> Same for me. I'm on an ipad pro (2020) with a jailbroken iOS 14.3. I tried both the cydia and sideloaded ipa binary. Makes no difference.

**Comment by Egleyser** [comment]
> > > because it doesn't close abruptly, like a bug or something, it just minimizes it. If I try to create a VM it starts and I can even see the boot screen, for example. However, importing the VM image will not.
> > 
> > 
> > Same for me. I'm on an ipad pro (2020) with a jailbroken iOS 14.3. I tried both the cydia and sideloaded ipa binary. Makes no difference.

Friend, sideload of utm do for Altstore with Altdaemon installed.

**Comment by tristan-k** [comment]
Can you explain why you are sure altstore is the cause because I'm not even using it.

**Comment by Egleyser** [comment]
In the tests I did and as I reported above, with your device with Jailbreak clicking directly from the official repo, app opens, works, but the VM (regardless of which S.O) does not start. If you sideloaded the UTM via Altstore / Altdaemon (tweak) and subscribe, it works. This way after the sideload by Altstore you can be with or without Jailbreak that the VM starts normally. I did this test on both my ipad 8gen and iphone 6s, both were like this and worked after installing by Altstore.

**Comment by tristan-k** [comment]
Thanks for explaining. Unfortunately Altstore is no solution for me because I'm not regularly using macOS. I installed the IPA with the help of the tweak 'AppSync Unified' and I see no reason why UTM should not be able run VMs if installed that way.

**Comment by Egleyser** [comment]
It is not a rule to have a Mac to use the Altstore, it is possible to install it using Windows, once the Altstore is installed you can use the Altdaemon tweak which will automatically sign without the need for you to go to a computer again, from then you install the UTM. I also do not see the sense that it does not work installing via repo, or sideloaded by other means, but it was the only way that it worked for me.

**Comment by tristan-k** [comment]
I think I will have to wait until the issue is acknowledged and fixed.

**Comment by 9chu** [comment]
> `qemu-system: -accel tcg,tb-size=32: allocate 33554432 bytes for jit buffer: Invalid argument` that's the issue. The JIT buffer by default takes your memory size (128M) and divides that by 4. Seems like this is too small here. Either increase the size (512M) or use the advanced setting for JIT buffer size and bump that up to something like 128
> 
> EDIT: Hmm, I was able to get it to boot fine with `-accel tcg,tb-size=32` so it may be something else wrong. Is it possible you have some jailbreak tweak installed that's preventing UTM from obtaining permissions to use JIT?

Same for me, installed UTM from Cydia and it crashes running any VM. The error log shows the `Invalid argument` too.
OS Version: 14.0
iPad Pro Gen2
Jailbroken by unc0ver 6.0.0
